bundles/org.openhab.binding.cloudrain/src/main/java/org/openhab/binding/cloudrain/internal/CloudrainConfig.java

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.cloudrain.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.cloudrain.internal.api.CloudrainAPIConfig;
+
+/**
+ * The Cloudrain account configuration class.
+ *
+ * @author Till Koellmann - Initial contribution
+ */
+@NonNullByDefault
+public class CloudrainConfig implements CloudrainAPIConfig {
+    /**
+     * The default timeout for connecting to the Cloudrain API (10 seconds)
+     */
+    public static final Integer DEFAULT_CONNECTION_TIMEOUT = 10;
+    /**
+     * The default interval for updating the irrigation status from the API (30 seconds)
+     */
+    public static final Integer DEFAULT_IRRIGATION_INTERVALL = 30;
+    /**
+     * The default interval for updating the zone status from the API (300 seconds)
+     */
+    public static final Integer DEFAULT_ZONE_INTERVALL = 300;
+    /**
+     * The default setting whether remaining seconds of active irrigations shall be updated in real time without API
+     * access (default: true)
+     */
+    public static final Boolean DEFAULT_REALTIME = true;
+    /**
+     * The default setting whether the irrigation status shall be updated at projected irrigation ends additionally to
+     * the regular polling (default: true)
+     */
+    public static final Boolean DEFAULT_UPDATE_AT_END = true;
+    /**
+     * The default setting whether the binding shall be used in test mode (default: false)
+     */
+    public static final Boolean DEFAULT_TEST_MODE = false;
+
+    private @Nullable String user;
+    private @Nullable String password;
+    private @Nullable String clientId;
+    private @Nullable String clientSecret;
+
+    private Integer connectionTimeout = DEFAULT_CONNECTION_TIMEOUT;
+    private Integer irrigationUpdateInterval = DEFAULT_IRRIGATION_INTERVALL;
+    private Integer zoneUpdateInterval = DEFAULT_ZONE_INTERVALL;
+    private Boolean realtimeUpdates = DEFAULT_REALTIME;
+    private Boolean updateAfterIrrigation = DEFAULT_UPDATE_AT_END;
```

Comment:
I would prefer to keep it configurable for the following reason: The update will only be a benefit for users who typically use scheduled irrigations without manual intervention. Some users might use the system in a rather manual way, though. In this case the expected irrigation end will most likely not be correct. Those users might turn off this setting in order to safe unnecessary API calls. There is not yet an information on API rate limiting, but I thought it might be a good idea to be flexible.
As this setting is an "advanced setting" with an appropriate default value most users will not cope with it anyway.